src/main/java/Sam.java

```diff
@@ -0,0 +1,203 @@
+import java.util.Scanner;
+
+import sam.parser.Parser;
+import sam.task.Deadline;
+import sam.task.Event;
+import sam.task.Task;
+import sam.task.TaskList;
+import sam.task.Todo;
+import sam.ui.Ui;
+
+/**
+ * Represents the main application class for Sam, a task management application.
+ * This class handles the main program loop, user input processing, and
+ * coordinates between different components like UI, Parser, TaskList, and Storage.
+ */
+public class Sam {
+    private TaskList tasks;
+    private Storage storage;
+    private Ui ui;
+
+    /**
+     * Constructor for Sam class.
+     */
+    public Sam() {
+        this.ui = new Ui();
+        this.storage = new Storage("./data/sam.txt");
+        this.tasks = new TaskList(storage.load());
+    }
+
+    /**
+     * Gets the welcome message for GUI.
+     * @return The welcome message string
+     */
+    public String getWelcomeMessage() {
+        return "Hello! I'm Sam\nWhat can I do for you?";
+    }
+
+    /**
+     * Processes user input and returns a response for GUI.
+     * @param input The user input string
+     * @return The response string
+     */
+    public String getResponse(String input) {
+        String[] parsed = Parser.parse(input.trim());
```

Comment:
Would renaming the String[] attribute from parsed to parsedInput improve overall readability? Since parsed is also used as an object elsewhere, this change might help distinguish the two and reduce ambiguity.
```suggestion
        String[] parsedInput = Parser.parse(input.trim());
```